cmake/Modules/FindComputeCpp.cmake

```diff
@@ -204,51 +216,76 @@ define_property(
   the command line so that it is seen by the compiler first. Enables
   non-standards-conformant SYCL code to compile with ComputeCpp."
 )
+define_property(
+  TARGET PROPERTY INTERFACE_COMPUTECPP_FLAGS
+  BRIEF_DOCS "Interface compile flags to provide compute++"
+  FULL_DOCS  "Set additional compile flags to pass to compute++ when compiling
+  any target which links to this one."
+)
+define_property(
+  SOURCE PROPERTY COMPUTECPP_SOURCE_FLAGS
+  BRIEF_DOCS "Source file compile flags for compute++"
+  FULL_DOCS  "Set additional compile flags for compiling the SYCL integration
+  header for the given source file."
+)
 
 ####################
-#   __build_spir
+#   __build_ir
 ####################
 #
 #  Adds a custom target for running compute++ and adding a dependency for the
 #  resulting integration header.
 #
-#  targetName : Name of the target.
-#  sourceFile : Source file to be compiled.
-#  binaryDir : Intermediate directory to output the integration header.
-#  fileCounter : Counter included in name of custom target. Different counter
-#       values prevent duplicated names of custom target when source files with the same name,
-#       but located in different directories, are used for the same target.
+#  TARGET : Name of the target.
+#  SOURCE : Source file to be compiled.
+#  BINARY_DIR : Intermediate directory to output the integration header.
+#  COUNTER : Counter included in name of custom target. Different counter
+#       values prevent duplicated names of custom target when source files with
+#       the same name, but located in different directories, are used for the
+#       same target.
 #
-function(__build_spir targetName sourceFile binaryDir fileCounter)
-
-  # Retrieve source file name.
-  get_filename_component(sourceFileName ${sourceFile} NAME)
+function(__build_ir)
+  set(options)
+  set(one_value_args
+    TARGET
+    SOURCE
+    BINARY_DIR
+    COUNTER
+  )
+  set(multi_value_args)
+  cmake_parse_arguments(SDK_BUILD_IR
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
+  )
+  get_filename_component(sourceFileName ${SDK_BUILD_IR_SOURCE} NAME)
 
-  # Set the path to the Sycl file.
-  set(outputSyclFile ${binaryDir}/${sourceFileName}.sycl)
+  # Set the path to the integration header.
+  set(outputSyclFile ${SDK_BUILD_IR_BINARY_DIR}/${sourceFileName}.sycl)
 
   # Add any user-defined include to the device compiler
```

Comment:
Unfortunately this won't work for every build script:

```cmake
add_executable(app main.cpp)
add_sycl_to_target(TARGET app BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} SOURCES main.cpp)
target_include_directories(app PRIVATE ${SOMETHING_INCLUDE_DIRS}) # won't find this
target_link_libraries(app PRIVATE Boost::boost) # or boost
```

The include directories must be gathered lazily with a generator expression to make `add_sycl_to_target()` work with property propagation through `target_link_libraries()`. The same is true for compile definitions.

Compile options are a special case for us because we are invoking a different compiler so we should not copy all the compiler flags automatically in my opinion (`-g` or `-mavx` come to mind).